skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,408 @@
+  val fileSuffix: String
+
+  fun screenClassName(featureName: String): String = "${featureName.replace(fileSuffix, "")}Screen"
+
+  fun generate(packageName: String, className: String): FileSpec
+
+  fun isTestComponent(): Boolean = false
+
+  fun writeToFile(directory: Path, packageName: String?, className: String) {
+    val baseDirectory =
+      if (isTestComponent()) Path.of(directory.pathString.replace("src/main", "src/test"))
+      else directory
+    val fileSpec = generate(packageName.orEmpty(), className)
+    Files.createDirectories(baseDirectory)
+    val stringWriter = StringWriter()
+    fileSpec.writeTo(stringWriter)
+    val generatedCode = stringWriter.toString().replace("public ", "")
+    File("${baseDirectory}/${className}${fileSuffix}.kt").writeText(generatedCode)
+  }
+}
+
+class CircuitScreen : CircuitComponent {
+  override val fileSuffix = "Screen"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /**
+     * Generate Circuit Screen class, eg:
+     *
+     * @Parcelize class ${NAME}Screen : Screen { data class State( val eventSink: Event.() -> Unit =
+     *   {}, ) : CircuitUiState
+     * @Immutable sealed interface Event : CircuitUiEvent }
+     */
+    val screenClassName = ClassName(packageName, screenClassName(className))
+    val eventInterfaceCN = screenClassName.nestedClass("Event")
+    val eventInterface =
+      TypeSpec.interfaceBuilder(eventInterfaceCN.simpleName)
+        .addAnnotation(IMMUTABLE)
+        .addModifiers(KModifier.SEALED)
+        .addSuperinterface(CIRCUIT_UI_EVENT)
+        .build()
+
+    val stateClass =
+      TypeSpec.classBuilder("State")
+        .addModifiers(KModifier.DATA)
+        .primaryConstructor(
+          FunSpec.constructorBuilder()
+            .addParameter(
+              ParameterSpec.builder(
+                  "eventSink",
+                  LambdaTypeName.get(
+                    receiver = eventInterfaceCN,
+                    returnType = Unit::class.asTypeName(),
+                  ),
+                )
+                .defaultValue("{}")
+                .build()
+            )
+            .build()
+        )
+        .addProperty(
+          PropertySpec.builder(
+              "eventSink",
+              LambdaTypeName.get(receiver = eventInterfaceCN, returnType = Unit::class.asTypeName()),
+            )
+            .initializer("eventSink")
+            .build()
+        )
+        .addSuperinterface(CIRCUIT_UI_STATE)
+        .build()
+    val typeSpec =
+      TypeSpec.classBuilder(screenClassName)
+        .addSuperinterface(SCREEN_INTERFACE)
+        .addAnnotation(PARCELIZE)
+        .addType(stateClass)
+        .addType(eventInterface)
+        .build()
+    return FileSpec.get(packageName, typeSpec)
+  }
+}
+
+class CircuitPresenter(
+  private val assistedInjection: AssistedInjectionConfig,
+  private val additionalCircuitInject: Set<ClassName> = setOf(),
+  private val noUi: Boolean = false,
```

Comment:
Instead of a double-negative, let's just have `ui: Boolean = true`

Reply:
updated https://github.com/slackhq/slack-gradle-plugin/pull/799/commits/b695d14d10124b133737548dde5d8e1449199c74